src/VideoStreaming/VideoReceiver.cc

```diff
@@ -1009,52 +1009,52 @@ VideoReceiver::_keyframeWatch(GstPad* pad, GstPadProbeInfo* info, gpointer user_
 
 void VideoReceiver::takeSnapshot() {
 #if defined(QGC_GST_STREAMING)
-    qDebug() << "Taking Snapshot";
+     qCDebug(VideoReceiverLog)  << "Taking Snapshot";
     if (!_videoSink) {
         return;
     }
 
-    qDebug() << "Trying to take a snapshot";
+     qCDebug(VideoReceiverLog) << "Trying to take a snapshot";
     GstSample *videobuffer = nullptr;
-    GstCaps *caps;
+    GstCaps *caps = nullptr;
     gint width, height;
     GstMapInfo map;
 
     // The video sink now is a GstBin that *contains* a sink.
     GstElement *realSink = gst_bin_get_by_name(GST_BIN(_videoSink), "qmlsink");
```

Comment:
What you did will work. But it expects that _videosink is GstBin (not necessary, if we support customization - it could be just GstElement) and it expects that GstBin contains element called "qmlsink" (it was called "qmlglsink" in my branch, IIRC) which is also not necessary to be true.

Proper way would be to install "last-sample" property at GstBin level (probably around this line https://github.com/mavlink/qgroundcontrol/blob/master/src/FlightDisplay/VideoManager.cc#L351) and then simply query last-sample as you did in your previous version.